### PR TITLE
IGNITE-10978: Obsolete testcases has been removed

### PR DESCRIPTION
**Pull request by kbolyand**

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/IgniteCacheInterceptorSelfTestSuite.java [hunk 1]
```diff
@@ -63,12 +63,6 @@
         GridTestUtils.addTestIfNeeded(suite,GridCacheInterceptorReplicatedSelfTest.class, ignoredTests);
         GridTestUtils.addTestIfNeeded(suite,GridCacheInterceptorReplicatedWithStoreSelfTest.class, ignoredTests);
 
-// TODO GG-11141.
-//        GridTestUtils.addTestIfNeeded(suite,GridCacheOnCopyFlagTxPartitionedSelfTest.class, ignoredTests);
-//        GridTestUtils.addTestIfNeeded(suite,GridCacheOnCopyFlagReplicatedSelfTest.class, ignoredTests);
-//        GridTestUtils.addTestIfNeeded(suite,GridCacheOnCopyFlagLocalSelfTest.class, ignoredTests);
-//        GridTestUtils.addTestIfNeeded(suite,GridCacheOnCopyFlagAtomicSelfTest.class, ignoredTests);
-
         GridTestUtils.addTestIfNeeded(suite,CacheInterceptorPartitionCounterRandomOperationsTest.class, ignoredTests);
```
Comment (dspavlov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/IgniteCacheInterceptorSelfTestSuite.java]: I suggest to delete unused classes from code, as well

Comment (kbolyand) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/IgniteCacheInterceptorSelfTestSuite.java]: Done

File: modules/core/src/test/java/org/apache/ignite/testsuites/IgniteCacheTestSuite3.java [hunk 2]
```diff
@@ -138,10 +138,6 @@
         //GridTestUtils.addTestIfNeeded(suite,GridCacheReplicatedTxMultiNodeBasicTest.class, ignoredTests);
         //GridTestUtils.addTestIfNeeded(suite,GridCacheReplicatedTxReadTest.class, ignoredTests);
 
-        // TODO GG-11141.
-//        GridTestUtils.addTestIfNeeded(suite,GridCacheDeploymentSelfTest.class, ignoredTests);
-//        GridTestUtils.addTestIfNeeded(suite,GridCacheDeploymentOffHeapSelfTest.class, ignoredTests);
```
Comment (kbolyand) [inline modules/core/src/test/java/org/apache/ignite/testsuites/IgniteCacheTestSuite3.java]: Class file is absent already.

File: modules/core/src/test/java/org/apache/ignite/testsuites/IgniteCacheTestSuite3.java [hunk 3]
```diff
@@ -138,10 +138,6 @@
...
-//        GridTestUtils.addTestIfNeeded(suite,GridCacheDeploymentOffHeapValuesSelfTest.class, ignoredTests);
```
Comment (kbolyand) [inline modules/core/src/test/java/org/apache/ignite/testsuites/IgniteCacheTestSuite3.java]: Class file is absent already.

File: modules/core/src/test/java/org/apache/ignite/testsuites/IgniteCacheTestSuite3.java [hunk 4]
```diff
@@ -138,10 +138,6 @@
         //GridTestUtils.addTestIfNeeded(suite,GridCacheReplicatedTxMultiNodeBasicTest.class, ignoredTests);
         //GridTestUtils.addTestIfNeeded(suite,GridCacheReplicatedTxReadTest.class, ignoredTests);
 
-        // TODO GG-11141.
-//        GridTestUtils.addTestIfNeeded(suite,GridCacheDeploymentSelfTest.class, ignoredTests);
```
Comment (kbolyand) [inline modules/core/src/test/java/org/apache/ignite/testsuites/IgniteCacheTestSuite3.java]: Class is removed according to issue reporter suggestion "these tests were testing peer class loading for cache entry (key/value). This case is not to be supported in the nearest future".

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/IgniteCacheInterceptorSelfTestSuite.java (hunk 1 above)
Comment (kbolyand) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/IgniteCacheInterceptorSelfTestSuite.java]: Classes did not contain test cases and have been removed:

1. GridCacheOnCopyFlagTxPartitionedSelfTest
2. GridCacheOnCopyFlagReplicatedSelfTest
3. GridCacheOnCopyFlagLocalSelfTest
4. GridCacheOnCopyFlagAtomicSelfTest

**Pull request closed without merging**